darwin/dataset/download_manager.py

```diff
@@ -464,14 +473,33 @@ def _fetch_multiple_files(path: Path, response: requests.Response) -> None:
         path = dir_path / filename
         response = requests.get(url, stream=True)
         if response.ok:
-            _write_file(path, response)
+            _write_file(path, response, transform_file_function)
         else:
             raise Exception(
                 f"Request to ({url}) failed. Status code: {response.status_code}, content:\n{get_response_content(response)}."
             )
 
 
-def _write_file(path: Path, response: requests.Response) -> None:
+def _write_file(path: Path, response: requests.Response, transform_file_function=None) -> None:
     with open(str(path), "wb") as file:
         for chunk in response:
             file.write(chunk)
+    if transform_file_function is not None:
+        transform_file_function(path)
+
+
+def _rg16_to_grayscale(path):
```

Comment:
feel like this helper function should live in a `utils.py` style file not here

Reply:
Why though? It's only used here. If we ever need it somewhere else, we could move it and make it more public then.

Reply:
ok, agreed its unlikely to be used elsewhere 👍